healthcare/healthcare/doctype/sample_collection/sample_collection.py

```
@@ -76,134 +76,178 @@ def on_cancel(self):
-							)
-							if observation:
-								comp["status"] = "Collected"
-								comp["collection_date_time"] = now_datetime()
-								comp["specimen"] = comp_obs_ref.get(j + 1) or comp_obs_ref.get(
-									obs.get("name")
-								)
+		for index, obs in enumerate(context.selected):
+			collect_row(context, index, obs)
+		update_child_status(context)
+		update_collection_status(context)
+	except Exception as exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)